backend/cmm_helpers.ml

```diff
@@ -1582,12 +1633,14 @@ let make_alloc_generic ?(scannable_prefix = Scan_all) ~mode set_fn dbg tag
     Cop (Calloc mode, Cconst_natint (hdr, dbg) :: args, dbg)
   else
     let id = V.create_local "*alloc*" in
-    let rec fill_fields idx = function
-      | [] -> Cvar id
-      | e1 :: el ->
+    let rec fill_fields idx args memory_chunks =
+      match args, memory_chunks with
+      | [], [] -> Cvar id
+      | e1 :: el, m1 :: ml ->
         Csequence
-          ( set_fn idx (Cvar id) (int_const dbg idx) e1 dbg,
-            fill_fields (idx + 1) el )
+          ( alloc_generic_set_fn (Cvar id) (int_const dbg idx) e1 m1 dbg,
+            fill_fields (idx + 1) el ml )
```

Comment:
I think this needs to offset by the field length in words, not 1

Reply:
That's a good point, I'll fix that. On the other, it's quite worrisome that there are no tests that fail, even with a bug like that (I guess that until we have some vec128s, we only have values that use exactly one word...)

Reply:
Yeah, it seems difficult to get unboxed vec128s into a closure at the moment since all the operations are on the boxed types (until #2965)

Reply:
I think we should contrive to write a test somehow.  Can we introduce a temporary Lambda primitive (or even the real one)?

Reply:
My failing test was in #2969; it indeed works with these changes

Reply:
Fixed